packages/p/pyincpp/xmake.lua

```diff
@@ -6,25 +6,57 @@ package("pyincpp")
     add_urls("https://github.com/chen-qingyu/pyincpp/archive/refs/tags/$(version).tar.gz",
              "https://github.com/chen-qingyu/pyincpp.git")
 
+    add_versions("v2.3.0", "e10640c03a6ae9365a299f57a68a5f2a873146b07f92fd33bc4f673e21d68072")
     add_versions("v1.6.1", "6a49657cb1f36f4fa400706a631c300771972015b4181f8cc352aa3de7e1a2a3")
     add_versions("v1.6.0", "1e8e4bfde447c439974180e206087b309f50ac0e24aeddf39d21d73fd7067368")
     add_versions("v1.4.1", "f3de3b5044a5c640811e87782264acbaf14697cd8c35bb21ddcd4de5664a60d0")
     add_versions("v1.3.3", "2689349de9faa35d8bbefddcc7d29d49308a2badd58961cc2b1a8f80c96d0823")
     add_versions("v1.3.2", "687148704f278c292962cffe1f440e5a4cc33f2a82f5e5a17b23aab88a282951")
 
+    -- Some old platforms cannot support the C++20 standard well
+    if on_check then
+        on_check("android", "macosx", function (package)
+            if package:version():ge("2.0.0") then
+                import("core.tool.toolchain")
+                if package:is_plat("android") then
+                    local ndk = toolchain.load("ndk", {plat = package:plat(), arch = package:arch()})
+                    local ndk_ver = ndk:config("ndk")
+                    assert(ndk_ver and ndk_ver > "r22", "package(pyincpp): need ndk version > r22 for android")
+                else
+                    assert(macos.version() > "macos-12", "package(pyincpp): need os version > 12 for macosx")
```

Comment:
macos.version():gt("12")

Reply:
or we can use package:check_cxxsnippets to check c++20, it will be better

Reply:
> macos.version():gt("12")

It seem doesn't work...

Reply:
> or we can use package:check_cxxsnippets to check c++20, it will be better

check_cxxsnippets is already `configs = {languages = "c++20"}`...

Reply:
> > or we can use package:check_cxxsnippets to check c++20, it will be better
> 
> check_cxxsnippets is already `configs = {languages = "c++20"}`...

那个是 on_test 里面的，安装后的事情了，我是说 on_check 里面提前检测 c++20 是否支持，跟 on_test没啥关系，不用搞太复杂的检测片段，弄个一小段就行了

Reply:
> > macos.version():gt("12")
> 
> It seem doesn't work...

xmake l macos.version 可以先看下返回值，它应该返回的是 semver 语义版本，不是字符串

Reply:
> > > macos.version():gt("12")
> > 
> > 
> > It seem doesn't work...
> 
> xmake l macos.version 可以先看下返回值，它应该返回的是 semver 语义版本，不是字符串

```
xmake l macos.version
error: cannot get the version of the current macos!
xmake l windows.version
error: cannot import module: windows.version, not found!
```
看不到……

Reply:
> > > or we can use package:check_cxxsnippets to check c++20, it will be better
> > 
> > 
> > check_cxxsnippets is already `configs = {languages = "c++20"}`...
> 
> 那个是 on_test 里面的，安装后的事情了，我是说 on_check 里面提前检测 c++20 是否支持，跟 on_test没啥关系，不用搞太复杂的检测片段，弄个一小段就行了

Ok。但是之前fail的case也是支持C++20的，但是没有实现C++20标准里面的东西……

Reply:
那就这个库依赖c++20 哪个特性，就仅仅检测这个

Reply:
> 那就这个库依赖c++20 哪个特性，就仅仅检测这个

Ok